src/proxyNetworkProvider.ts

```diff
@@ -139,8 +151,14 @@ export class ProxyNetworkProvider implements INetworkProvider {
         return hashes;
     }
 
-    async simulateTransaction(tx: ITransaction): Promise<any> {
-        let response = await this.doPostGeneric("transaction/simulate", tx.toSendable());
+    async simulateTransaction(tx: ITransaction | ITransactionNext): Promise<any> {
+        if ("toSendable" in tx){
+            const response = await this.doPostGeneric("transaction/simulate", tx.toSendable());
+            return response;
+        }
+        
+        const transaction = prepareTransactionNextForBroadcasting(tx);
```

Comment:
A comment below - perhaps have the preparation logic extracted in `transactions.ts` - to handle both, not only "next".

Reply:
Done!